src/main/kotlin/com/zepben/evolve/streaming/grpc/GrpcChannelBuilder.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Zeppelin Bend Pty Ltd
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package com.zepben.evolve.streaming.grpc
+
+import com.zepben.auth.client.ZepbenTokenFetcher
+import io.grpc.CallCredentials
+import io.grpc.ChannelCredentials
+import io.grpc.TlsChannelCredentials
+import io.grpc.netty.shaded.io.grpc.netty.NettyChannelBuilder
+import java.io.File
+
+/**
+ * Builder class for GrpcChannel. Allows easy specification of channel credentials via SSL/TLS
+ * and call credentials via a ZepbenTokenFetcher.
+ */
+class GrpcChannelBuilder {
+
+    private var _host: String = "localhost"
+    private var _port: Int = 50051
+    private var _channelCredentials: ChannelCredentials? = null
+    private var _callCredentials: CallCredentials? = null
+
+    fun build(): GrpcChannel = GrpcChannel(
+        _channelCredentials?.let { channelCreds ->
+            val channelBuilder = NettyChannelBuilder.forAddress(_host, _port, channelCreds)
+            _callCredentials?.let { callCreds ->
+                channelBuilder.intercept(CallCredentialApplier(callCreds)).build()
+            } ?: channelBuilder.build()
+        } ?: NettyChannelBuilder.forAddress(_host, _port).build()
+    )
+
+    fun forAddress(
+        host: String,
+        port: Int
+    ): GrpcChannelBuilder {
+        _host = host
+        _port = port
+        return this
+    }
+
+    fun makeSecure(
+        rootCertificates: File? = null,
+        privateKey: File? = null,
+        certificateChain: File? = null
+    ): GrpcChannelBuilder {
+        var channelCredentialsBuilder = TlsChannelCredentials.newBuilder()
+        if (rootCertificates != null) {
+            channelCredentialsBuilder = channelCredentialsBuilder.trustManager(rootCertificates)
+        }
+        if (privateKey != null && certificateChain != null) {
+            channelCredentialsBuilder = channelCredentialsBuilder.keyManager(certificateChain, privateKey)
+        }
+        _channelCredentials = channelCredentialsBuilder.build()
+        return this
+    }
```

Comment:
@kgreav what is the purpose of private key and certificate chain if the client doesn't serve the TLS connection? Isn't the identity of the client confirmed on the application layer via auth0?

Reply:
The server could have client verification enabled, in which case the server verifies the client has a cert signed by a CA the server trusts. It can be used in conjunction with Auth0 or on its own. If you're already using Auth0 the benefit is minimal, but still exists.

It's something seldom used but should be supported.

Reply:
Should I add the option to use client verification in the connect methods or is the use case too limited for that?

Reply:
Maybe just add functionality to the GrpcChannelBuilder to do it but don't bother with the connect functions? That way a user can still do it themselves but it keeps the connect functions a bit more concise.